### ISSUE TITLE
`env xonsh` from within Xonsh directly backgrounds certain graphical apps

### ISSUE THREAD

**Issue by sevens-ef** [xonsh/xonsh#3103]
## xonfig

<details>

```
$ xonfig
+------------------+----------------------+
| xonsh            | 0.8.12               |
| Git SHA          | 986bf28d             |
| Commit Date      | Mar 11 14:37:52 2019 |
| Python           | 3.6.7                |
| PLY              | 3.11                 |
| have readline    | True                 |
| prompt toolkit   | 2.0.9                |
| shell type       | prompt_toolkit2      |
| pygments         | 2.3.1                |
| on posix         | True                 |
| on linux         | True                 |
| distro           | ubuntu               |
| on darwin        | False                |
| on windows       | False                |
| on cygwin        | False                |
| on msys2         | False                |
| is superuser     | False                |
| default encoding | utf-8                |
| xonsh encoding   | utf-8                |
| encoding errors  | surrogateescape      |
+------------------+----------------------+
```
</details>

Though I've seen the same issue in earlier Xonsh versions (at least in 0.8.x) and with earlier Python 3.6 versions (and on different distros as well).

## Expected Behavior
A nested Xonsh using `env xonsh` to behave the same as a regular Xonsh session.

## Current Behavior
When starting a nested Xonsh via `env xonsh`, many programs that try to draw to the screen are immediately sent to the background (no drawings were ever visible, `^Z` is printed to the shell and a new prompt is shown); e.g. running `man`, `vim` or `less` (though at least `tmux` and `screen` seem fine).  Running the same commands in a `env bash` launched shell works fine.

Starting a nested Xonsh via plain `xonsh` also works fine.

## Steps to Reproduce

```
xonsh --no-rc
# Works fine
man man

env xonsh --no-rc
# Prints `^Z`, returns to prompt, shows `man man` in `jobs`
man man
```

For completeness: starting `env xonsh` from a Bash session works fine, and starting a nested Xonsh as `env -u PAGER -u MANPAGER xonsh --no-rc` seems to make no difference.

I can fix the behavior by marking `env` as not threadable in my `.xonshrc`.

May be related to #2759 (at least the same workaround fixes both issues).

**Comment by jb-leger** [comment]
Should be fixed by PR #3106. @sevens-ef can you confirm that it is fixed now?

**Comment by sevens-ef** [comment]
I can (somewhat) confirm that it's fixed: opening `man`, `less` and `vim` from a `env xonsh` launched from within a Xonsh session works in Xonsh 0.9.4, whereas it doesn't in 0.9.3.

With the launcher script that originally triggered the problem it's not working though, but even my workaround doesn't work with this in Xonsh 0.9.x anymore (and it has various other problems with 0.9.x) so that's probably due to something unrelated.  Will reopen/open a new issue if it turns out to be related.  N.b.: manually running the exact same `env` command as the script does to launch Xonsh also has it fixed.
